### PR TITLE
[ENH] Reverted hdf5 functions and added to_dict method

### PR DESCRIPTION
**Pull request by raj1701**
- Reverted read/write hdf5 functions and their associated tests
- Added warnings for read/write using txt format and associated tests
- Added to_dict method

### CODE REVIEW

**Review by jasmainak — APPROVED** [review]
Looks fine to me. Anything to revert in `whats_new`? @ntolley or @rythorpe can probably take a second look !

File: hnn_core/cell_response.py [hunk 1]
```diff
@@ -402,30 +400,31 @@ def plot_spikes_hist(self, trial_idx=None, ax=None, spike_types=None,
...
-
-    def _write_txt(self, fname):
+        cell_response_data = dict()
+        cell_response_data['spike_times'] = self.spike_times
+        cell_response_data['spike_gids'] = self.spike_gids
+        cell_response_data['spike_types'] = self.spike_types
+        vsec_data = self.vsec
+        cell_response_data['vsec'] = list()
+        for trial in vsec_data:
+            trial = dict((str(key), val) for key, val in trial.items())
```
Comment (ntolley) [inline hnn_core/cell_response.py:417]:
```suggestion
            # Turn `int` gid keys into string values for hdf5 format
            trial = dict((str(key), val) for key, val in trial.items())
```

File: hnn_core/cell_response.py [hunk 2]
```diff
@@ -402,30 +400,31 @@ def plot_spikes_hist(self, trial_idx=None, ax=None, spike_types=None,
...
+            cell_response_data['vsec'].append(trial)
+        isec_data = self.isec
+        cell_response_data['isec'] = list()
+        for trial in isec_data:
+            trial = dict((str(key), val) for key, val in trial.items())
```
Comment (ntolley) [inline hnn_core/cell_response.py:422]:
```suggestion
            # Turn `int` gid keys into string values for hdf5 format
            trial = dict((str(key), val) for key, val in trial.items())
```

**Comment by rythorpe** [comment]
Sorry I'm kinda out of the loop here. Is the idea to be able to convert everything to dict objects so that they can be stored as hdf5 using an external function?

**Comment by raj1701** [comment]
Yes @rythorpe. I've added similar `to_dict()` functions for `cell`, `section` and `extracellular array`

**Comment by rythorpe** [comment]
Okay cool, and then the `write_hdf5(cell_response)` will be called where?

**Comment by raj1701** [comment]
Cell response can only be saved along the network as per the latest discussion therefore this code will be shifted in `network.py`

**Comment by ntolley** [comment]
Ok so I looked into link check problem and it is something on the Brown U server side. Hopefully it'll be taken care of soon, but we might consider setting the link check to temporarily ignore brown.edu hosted websites if it keeps this PR from being merged much longer.

**Comment by rythorpe** [comment]
> Cell response can only be saved along the network as per the latest discussion therefore this code will be shifted in `network.py`

Got it, thanks!

**Comment by jasmainak** [comment]
I went ahead and merged ... hopefully linkcheck will be resolved on its own. Thanks @raj1701 !

**Pull request merged**